### PR TITLE
docs: use `@publicApi` tags

### LINKED ISSUE #21661: Remove experimental flag from stable APIs
A lot of our APIs stay flagged as "experimental" because we never check if we should change it.
Before each major release, we should ask each component owner to remove the experimental flag from APIs that have been around for more than 1 version.
For example almost everything in i18n is flagged experimental.

@StephenFluin @IgorMinar what do you think about that?

Task breakdown:

- [ ] remove @experimental and @stable from source code
- [ ] update API golden files
- [ ] remove stable and experimental badges from API templates
- [ ] remove "experimental" from the API filter

### PR DESCRIPTION
**Pull request by petebacondarwin**
By the way, the `@publicApi` tags are not title cased, as github would have you believed in the commit messages.

Closes #21661

### CODE REVIEW

File: packages/platform-server/src/utils.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ export function renderModule<T>(
  * `url` is the URL for the current render request.
  * `extraProviders` are the platform level providers for the current render request.
  *
- * @experimental
+ * @publicApi
```
Comment (IgorMinar) [inline packages/platform-server/src/utils.ts]: this is a method and not a top level symbol. could we ban these tags on non-top-level symbols? (maybe as a follow up PR?)

File: tools/ts-api-guardian/lib/cli.ts [hunk 2]
```diff
@@ -24,9 +24,9 @@ export function startCli() {
   const options: SerializationOptions = {
     stripExportPattern: [].concat(argv['stripExportPattern']),
     allowModuleIdentifiers: [].concat(argv['allowModuleIdentifiers']),
-    exportTags: {required: [], banned: [], toCopy: ['deprecated', 'experimental']},
-    memberTags: {required: [], banned: [], toCopy: ['deprecated', 'experimental']},
-    paramTags: {required: [], banned: [], toCopy: ['deprecated', 'experimental']}
+    exportTags: {required: ['publicApi'], banned: ['experimental'], toCopy: ['deprecated']},
+    memberTags: {required: [], banned: ['experimental'], toCopy: ['deprecated']},
```
Comment (IgorMinar) [inline tools/ts-api-guardian/lib/cli.ts]: oh look, we could add publicApi in the banned category for member and paramTags! :)

**Review by IgorMinar — APPROVED** [review]
lgtm. I kicked off a presubmit to ensure that we are not breaking closure compiler or tsickle with this new tag:         http://test/OCL:217902220:BASE:217902921:1539973934332:990eaed4

**Comment by IgorMinar** [comment]
can you please fix the lint CI job too?

File: packages/platform-server/src/utils.ts (hunk 1 above)
Comment (petebacondarwin) [reply packages/platform-server/src/utils.ts]: I don't think it is a method...

File: tools/ts-api-guardian/lib/cli.ts (hunk 2 above)
Comment (petebacondarwin) [reply tools/ts-api-guardian/lib/cli.ts]: 👍

**Comment by IgorMinar** [comment]
g3 is good

**Pull request closed without merging**